ansible_collections/arista/avd/roles/eos_designs/templates/network_services/logic.j2

```diff
@@ -22,10 +22,25 @@
 {%             if switch.filter_tenants is arista.avd.contains([tenant.name, "all"]) %}
 {#                 vrfs #}
 {%                 set tmp_vrfs = [] %}
+{%                 set tmp_bgp_peer_groups = [] %}
+{%                 set tmp_mapped_peer_groups = [] %}
 {%                 for vrf in tenant.vrfs | arista.avd.convert_dicts | arista.avd.natural_sort('name') %}
+{%                     set tmp_mapped_vrf_peer_groups = [] %}
 {%                     do vrf.update({'bgp_peers': vrf.bgp_peers | arista.avd.convert_dicts('ip_address') | arista.avd.natural_sort('ip_address')}) %}
+{%                     for bgp_peer in vrf.bgp_peers %}
+{%                         if (bgp_peer.peer_group is arista.avd.defined) and (inventory_hostname in bgp_peer.nodes | arista.avd.default([])) and (bgp_peer.peer_group not in tmp_mapped_vrf_peer_groups) %}
+{%                             do tmp_mapped_vrf_peer_groups.append(bgp_peer.peer_group) %}
+{%                             if bgp_peer.peer_group not in tmp_mapped_peer_groups %}
+{%                                 do tmp_mapped_peer_groups.append(bgp_peer.peer_group) %}
+{%                             endif %}
+{%                         endif %}
+{%                     endfor %}
```

Comment:
We can just accept duplicates in the list, since we only use if for the `if in` test.
```suggestion
{%                     set tmp_mapped_vrf_peer_groups = vrf.bgp_peers | selectattr("peer_group", "arista.avd.defined") | map(attribute="peer_group") %}
{%                     do tmp_mapped_peer_groups.extend(tmp_mapped_vrf_peer_groups) %}
```

Reply:
updated suggestion